# Chapter 1

In [1]:
import pandas as pd

wage1 = pd.read_stata("stata/WAGE1.DTA")

In [2]:
#Question C1.i
print(wage1.educ.mean())
print(wage1.educ.min(), wage1.educ.max())
#Question C2.ii
print(wage1.wage.mean())

12.562737642585551
0 18
5.896100044250488


C1.i: Average education is 12.56, lowest is 0, highest is 18

C1.ii: Average hourly wage is \\$5.90, this seems low since state minimum wages seem to be approximately \\$7.00

C1.iii: CPI from 1976 is 56.9, for 2010 is 218.056

In [3]:
#CPI in 1976 is 56.9 (average), CPI in 2010 is 218.056
(wage1.wage * ((2.18056 / 0.569))).mean()

22.595443725585938

C1.iv: Average hourly wage is now \\$22.60 which is more consistent with reported averages

In [4]:
print(wage1.female.sum())
print(wage1.shape[0] - wage1.female.sum())

252
274


C1.v: 252 observations are women, 274 are men.

In [5]:
#Question C2
bwght = pd.read_stata("stata/BWGHT.DTA")

In [6]:
print(bwght.shape)

(1388, 14)


In [7]:
bwght.cigs.mean()

2.0871757925072045

In [8]:
bwght.cigs.loc[bwght.packs > 0].mean()

13.665094339622641

In [9]:
bwght.fatheduc.mean()

13.186241610738255

In [10]:
print(bwght.faminc.mean())
print(bwght.faminc.std())

29.026657104492188
18.739294052124023


C2.i: 1388 women in the sample

C2.ii: 2.09 cigarettes per day on average. This is not a good measure of 'average' women since most do not smoke at all

C2.iii: 13.67 cigarettes per day. This is higher than in (ii) since we are measuring the average of packs smoked by smokers (not weighted down by 0s)

C2.iv: The mean of fatheduc is 13.19 but ~200 values are missing meaning they are not included in the calculation

C2.v: Average family income is \\$29,026 with a standard deviation of \$18,739

In [11]:
# Question 3
meap01 = pd.read_excel("excel/meap01.xls", 
                     names = ["dcode", "bcode", "math4", "read4", "lunch", "enroll", "expend", "exppp", "lenroll", "lexpend", "lexppp"],
                     sep = " ",
                     header = None)

In [12]:
print(meap01.math4.min())
print(meap01.math4.max())

0.0
100.0


In [13]:
print(meap01.math4.loc[meap01.math4 == 100.0].size)
print(meap01.math4.loc[meap01.math4 == 50.0].size)
print(meap01.shape[0])

38
17
1823


In [14]:
print(meap01.math4.mean())
print(meap01.read4.mean())

71.90899616017558
60.061876028524416


In [15]:
meap01.math4.corr(meap01.read4)

0.8427281443588261

In [16]:
meap01.exppp.std()

1091.8895895804048

In [17]:
(6000 / 5500 - 1) * 100

9.090909090909083

In [18]:
import numpy as np
100 * (np.log(6000) - np.log(5500))

8.701137698962924

C3.i Largest = 100%, smallest = 0%

C3.ii 38 schools had a perfect pass rate, about 2%

C3.iii 17 schools have a pass rate of exactly 50%

C3.iv Reading is a more difficult test to pass

C3.v Correlation between reading and math is 0.84. Reading and math are highly correlated

C3.vi The standard deviation is about \\$1,000 compared to an average of about \$5,000

In [19]:
jtrain2 = pd.read_stata("stata/JTRAIN2.DTA")

In [20]:
jtrain2.train.sum() / jtrain2.shape[0]

0.4157303370786517

In [21]:
print(jtrain2.re78[jtrain2.train == 0].mean())
print(jtrain2.re78[jtrain2.train == 1].mean())

4.554801940917969
6.349145412445068


In [22]:
print(jtrain2.unem78[jtrain2.train == 1].sum())
print(jtrain2.unem78[jtrain2.train == 0].sum())
print(jtrain2[jtrain2.train == 0].shape[0])

45
92
260


C4.i 41.6% of the men in the sample recieved job training

C4.ii Men in the training program recieved, on average, \\$6,349 compared to the average of \$4,555 for those not training. This is large.

C4.iii About 24% of the men trained were unemployed compared to 35% unemployment for men not getting job training. Unemployed men are overrepresented in the untrained.

C4.iv It would appear the job training is effective until the difference in unemployment is noticed. The result would need to be normalized for unemployment before being convincing.

In [23]:
fertil2 = pd.read_stata("stata/FERTIL2.DTA")

In [24]:
print(fertil2.children.min())
print(fertil2.children.max())
print(fertil2.children.mean())

0
13
2.2678284797064894


In [25]:
fertil2.electric.sum() / fertil2[fertil2.electric != np.nan].shape[0]

0.14010548039440496

In [26]:
print(fertil2.children[fertil2.electric == 1].mean())
print(fertil2.children[fertil2.electric == 0].mean())

1.8985270049099836
2.327728849746464


C5.i Smallest number of children = 0, largest = 13

C5.ii 14% of women have electricity at home

C5.iii On average there are 1.9 children in homes with electricity, and 2.33 in homes without

C5.iv We cannot conclude that having electricity causes more women to have fewer children. There may be an income effect to consider (low income families tend to have a larger number of children, as well as being less likely to have electricity) for starters.